protos/pointer-draft/pointer.proto

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+syntax = "proto3";
+
+import "google/protobuf/duration.proto";
+
+message RedundancyScheme {
+  enum SchemeType {
+    RS = 1;
+  }
+  SchemeType type = 1;
+
+  // these values apply to RS encoding
+  int64 required = 2;  // the minimum required for reconstruction
+  int64 total = 3;     // the total amount of pieces we generated
+  int64 repair_threshold = 4;  // the amount of pieces we need to drop to before triggering repair
+  int64 success_threshold = 5;  // the amount of pieces we need to store to call it a success
+}
+
+message EncryptionScheme {
+  enum EncryptionType {
+    AESGCM = 1;
+    SECRETBOX = 2;
+    // only allow authenticated encryption schemes
+  }
+  EncryptionType type = 1;
+  bytes encrypted_encryption_key = 2;
+  bytes encrypted_starting_nonce = 3;
+}
+
+message RemotePiece {
+  int64 piece_num = 1;
+  string node_id = 2;
+  bool parity = 3;
+  int64 size = 4;  // full size, including merkle tree
+}
+
+message InlineSegment {
+  bytes segment = 1;
+}
+
+message RemoteSegment {
+  RedundancyScheme redundancy = 1;
+  bytes piece_name = 2;
+  repeated RemotePiece remote_pieces = 3;
+  repeated Farmer farmers = 4; // list of farmers
```

Comment:
remote_pieces is actually the same thing as the farmer list. a remote piece has a node_id. we don't need both so we can just pick one